ts/src/profiler.ts

```diff
@@ -126,6 +142,46 @@ async function profileBytes(p: perftools.profiles.IProfile): Promise<string> {
 }
 
 /**
+ * Error constructed from http server response which indicates backoff.
+ */
+class BackoffResponseError extends Error {
+  // tslint:disable-next-line: variable-name
+  __proto__: Error;
```

Comment:
I don't understand what's happening here. Can you explain what you're trying to do?

Reply:
instanceof doesn't seem to work for classes that extend Error. I did what this issue suggested: https://github.com/Microsoft/TypeScript/issues/13965.

Reply:
A few points
* In case this this where the code is not understandable without the background, you should add a link and explain why the code looks wonky.
* Because of the wonkiness, I would recommend a TypeScript type guard instead of instanceof in the first place.

Reply:
done.